### PR TITLE
[IOTDB-3167] Add wait_leader_ready logic for RatisConsensus

### PR DESCRIPTION
**Pull request by OneSizeFitsQuorum**
see [jira3167](https://issues.apache.org/jira/browse/IOTDB-3167)

### CODE REVIEW

File: consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java [hunk 1]
```diff
@@ -209,18 +216,20 @@ public ConsensusWriteResponse write(
...
+      return failedRead(new ConsensusGroupNotExistException(consensusGroupId));
     }
 
     RaftClientReply reply;
     try {
       RequestMessage message = new RequestMessage(IConsensusRequest);
       RaftClientRequest clientRequest =
-          buildRawRequest(groupId, message, RaftClientRequest.staleReadRequestType(-1));
+          buildRawRequest(
+              groupId, message, RaftClientRequest.staleReadRequestType(getCommitIndex(groupId)));
```
Comment (SzyWilliam) [inline consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java:232]: stale_read seems to read the most-updated data by default, which is determined by commit index. Assign a commit index here may be unnecessary?

**Review by SzyWilliam — APPROVED** [review]

**Review by CRZbulabula — APPROVED** [review]
LGTM.

File: consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java (hunk 1 above)
Comment (OneSizeFitsQuorum) [reply consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java:232]: In this way, data can be read after the restart is complete

**Comment by OneSizeFitsQuorum** [comment]
Has been merged in [pr5904](https://github.com/apache/iotdb/pull/5904)

**Pull request closed without merging**